### PR TITLE
arch: xtensa: merge UP and SMP architectures

### PR DESCRIPTION
**Pull request by tlauda**
Merges xtensa UP and SMP architectures.
The amount of code which were duplicated by creating
SMP architecture as separate instance is too big
to maintain. I've decided to merge both architectures,
change explicit selection of SMP to be dependent on
configured number of cores (> 1) and add branching in
xtos based on currently used processor configuration.

Signed-off-by: Tomasz Lauda <tomasz.lauda@linux.intel.com>

### CODE REVIEW

File: src/arch/xtensa/CMakeLists.txt [hunk 1]
```diff
@@ -38,19 +38,11 @@ set(platform_bootldr_ld_script boot_ldr.x)
...
-		${PROJECT_SOURCE_DIR}/src/arch/xtensa/smp/xtos
-	)
-else()
-	target_include_directories(sof_options INTERFACE
-		${PROJECT_SOURCE_DIR}/src/arch/xtensa/up/include
-		${PROJECT_SOURCE_DIR}/src/arch/xtensa/up/xtos
-	)
-endif()
+target_include_directories(sof_options INTERFACE
+	${PROJECT_SOURCE_DIR}/src/arch/xtensa/hal
```
Comment (jajanusz) [inline src/arch/xtensa/CMakeLists.txt]: Why is `hal` in include dirs? I don't see any headers there.

File: src/arch/xtensa/include/arch/alloc.h [hunk 2]
```diff
@@ -6,14 +6,17 @@
...
+ * \brief Xtensa memory allocation header file
  * \authors Tomasz Lauda <tomasz.lauda@linux.intel.com>
  */
 
 #ifndef __ARCH_ALLOC_H__
 #define __ARCH_ALLOC_H__
 
+#include <config.h>
+
+#if CONFIG_SMP
```
Comment (jajanusz) [inline src/arch/xtensa/include/arch/alloc.h]: Isn't `CONFIG_SMP` removed at all? I see it being removed from `src/arch/xtensa/Kconfig`, but maybe I missed that it's moved somewhere else.

Comment (jajanusz) [reply src/arch/xtensa/include/arch/alloc.h]: Nvm, I see it's just now helper config.

Comment (paulstelian97) [reply src/arch/xtensa/include/arch/alloc.h]: It wasn't removed in the first patch but moved below the CORE_COUNT config, and its default value is now saner (defaults to SMP build when CORE_COUNT > 1).

Comment (jajanusz) [reply src/arch/xtensa/include/arch/alloc.h]: @paulstelian97 Yea, i noticed it later ;p

**Pull request merged**